drpcconn/conn.go

```diff
@@ -112,6 +125,18 @@ func (c *Conn) NewStream(ctx context.Context, rpc string) (_ drpc.Stream, err er
 		return nil, err
 	}
 
+	invokeMsg := make([]byte, 0)
+	metadata, ok := drpcmetadata.Get(ctx)
+	if ok {
+		invokeMsg, err = metadata.Encode(invokeMsg)
+		if err != nil {
+			return nil, err
+		}
+		if err := stream.RawWrite(drpcwire.KindInvokeMetadata, invokeMsg); err != nil {
+			return nil, err
+		}
+	}
+
```

Comment:
Same as above: please just have `doNewStream` take the metadata byte slice. Perhaps something like:

```go
var metadata []byte
if md, ok := drpcmetadata.Get(ctx); ok {
    metadata, err = md.Encode(nil)
    if err != nil {
        return nil, err
    }
}

if err := c.doNewStream(stream, []byte(rpc), metadata); err != nil {
...
```